resources/design-document.md

```diff
@@ -1,75 +1,222 @@
+String name
+String category
+String regionOfOrigin
+int regionalDemand
+double salesForecast
+double perPallet
+double weight
+double purchaseCost
+double baseMargin
+double rateOfReplenishment
+String synergy
+boolean active
+```
+
+```
+//TransactionModel
+transactionId
+String warehouseId
+String shipmentId
+String itemId
+int count
+Date date
+String partnerId
+String transactionType
+
+```
+
+## 6.2. Create Item Endpoint
+
+* Accepts POST requests to /items
+* Accepts input data and user info to create a new item and returns an itemModel for the logged-in user
+
+## 6.3 Get Item Endpoint
+* Accepts GET requests to /items/:itemId
+* Accepts an itemId and user info and returns the relevant itemModel
+
+## 6.4 Get All Items Endpoint
+* Accepts GET requests to /items
+* Accepts user info and return a list of all Items
+
+## 6.5 Update Item Endpoint
+* Accepts PUT requests to /items/:itemId
+* Accepts an itemId, input data, and user info, updates the item data, and returns an itemModel
+
+## 6.6 Create Warehouse Endpoint
+* Accepts POST requests to /warehouses
+* Accepts input data and user info to create a new warehouse
+* Returns the created warehouseModel
+
+## 6.7 Get Warehouse Endpoint
+* Accepts GET requests to /warehouses/:warehouseId
+* Accepts a warehouseId and user info, and returns the relevant warehouseModel
+
+## 6.8 Get All Warehouses Endpoint
+* Accepts GET requests to /warehouses
+* Accepts user information and returns all warehouses associated with the account
+
+## 6.9 Delete Warehouse Endpoint
+* Accepts DELETE requests to /warehouses/warehouseId
+* Accepts a warehouseId and user info, and deletes the matching warehouse, and shipping data
+
+## 6.10 Update Warehouse Endpoint
+* Accepts PUT requests to /warehouses/:warehouseId
+* Accepts a warehouseId, allows the user to change the warehouse name, and returns the warehouseModel
+* 
+## 6.11 Update Warehouse Inventory Endpoint
+* Accepts PUT requests to /warehouses/inventory/:warehouseId
+* Accepts a warehouseId, itemIds in a map of stock change amounts for each item, a partnerID, an optional date field, and user info
+* Generates a transactionModel object and adds it to the transactions table for reporting
+* Updates warehouse inventory count and returns warehouseModel
+
+## 6.12 FillWarehouseLambda
+* Accepts PUT requests to /warehouses/fill/:warehouseId
+* Accepts a warehouseId
+* If the warehouse is empty, retrieves a list of all items and calculates the most profitable selection for the given warehouse metrics
+* returns an updated warehouseModel
+
+## 6.13 Get WareHouse Inventory Report
+* Accepts GET requests to warehouses/:warehouseId/inventory
+* Accepts a warehouseId and returns a warehouseModel
+* Calculates current inventory status for current warehouse inventory items, and pulls item data
+* Calculates item data and recommends new items to be added to the warehouse, and items to be removed
+* Returns a JSON formatted report to be formatted in the front end
```

Comment:
I see the mock-ups have a Warehouse page (that I imagine shows the current inventory) and a separate Warehouse Inventory Report page. Are you envisioning those both using this endpoint? If so, how would you distinguish whether or not to generate/include the recommendations?

Reply:
What I'm envisioning here is that I would use the Get Warehouse endpoint to display the inventory contents on the Warehouse page on page load (or possibly on a button press depending on the limits of lambda and what I can do on the front end).  This page would be used to view the current inventory and adjust the quantity of each item, which will update the warehouse and generate transactions for reports.  

 The Get Warehouse Inventory Report endpoint would be a separate page that would show the inventory in a more readable report format have a section recommending inventory changes (and possibly some other relevant data if I can think of anything useful).  Part of my idea with these reports is that I  would like to interface with a third party API and generate pdf documents of the reports, fulfilling the third-party integration requirement of the project even if I can't get to my stretch goals.